### PR TITLE
Extend filecheck module with dir size chart

### PR DESCRIPTION
**Pull request by ski2per**
Signed-off-by: Ted <ski2per@163.com>

Hi, it's me again. I create this PR for this [#486 ](https://github.com/netdata/go.d.plugin/pull/486).
Just move the file/dir calculation func into filecheck module.

### CODE REVIEW

**Comment by ilyam8** [comment]
Hi @ski2per 

i think we need to make dir size calculating optional (it could consume a lot of cpu if there are a lot of files) and likely use other walk dir library (ex.: https://github.com/karrick/godirwalk), but we can do it in another PR.

**Comment by ilyam8** [comment]
And seems like we need an `recursive` option - `calDirNumOfFiles` returnes num of files in the root dir, `calcDirSize` does it recursively 🤔 On the other hand `numOfFiles` includes number of dirs, so it is kinda ok i think.

**Review by ilyam8 — APPROVED** [review]

File: modules/filecheck/collect_dirs.go [hunk 1]
```diff
@@ -59,6 +63,10 @@ func (fc *Filecheck) addDirToCharts(dirpath string) {
 			id = dirDimID(dirpath, "mtime_ago")
 		case dirNumOfFilesChart.ID:
 			id = dirDimID(dirpath, "num_of_files")
+		case dirNumOfFilesChart.ID:
+			id = dirDimID(dirpath, "num_of_files")
```
Comment (vlvkobal) [inline modules/filecheck/collect_dirs.go:67]: A duplicate. Please remove it.

Comment (ilyam8) [reply modules/filecheck/collect_dirs.go:67]: hawk eye! 🐦

Comment (ilyam8) [inline modules/filecheck/collect_dirs.go:67]:
```suggestion
```

Comment (ilyam8) [reply modules/filecheck/collect_dirs.go]: removed in https://github.com/netdata/go.d.plugin/pull/487/commits/edc7b2f627bdd88d158468c04493ca3b13af7285

**Review by vlvkobal — APPROVED** [review]

**Comment by ilyam8** [comment]
Many thanks @ski2per !

**Comment by ski2per** [comment]
> Hi @ski2per
> 
> i think we need to make dir size calculating optional (it could consume a lot of cpu if there are a lot of files) and likely use other walk dir library (ex.: https://github.com/karrick/godirwalk), but we can do it in another PR.

I have thought this before, I will do it in another PR.

**Comment by ski2per** [comment]
> Many thanks @ski2per !

You are welcome 😄

**Comment by ilyam8** [comment]
i opened an issue https://github.com/netdata/go.d.plugin/issues/488

As for library don't hurry with the PR - i'll open an issue and we need to do benchmarks first.

**Pull request merged**